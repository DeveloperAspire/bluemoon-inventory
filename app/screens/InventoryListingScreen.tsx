import { StyleSheet } from "react-native";
import React from "react";
import InventoryItem from "../components/common/InventoryItem";
import Screen from "../components/Screen";

const InventoryListingScreen = ({
  navigation,
}: {
  navigation: Record<string, any>;
}) => {
  return (
    <Screen style={styles.container}>
      <InventoryItem
        name="Bags of senegal beans"
        totalStock={3}
        price={60000}
        description="Bags of senegal beans due for delivery and sale"
        onPress={() =>
          navigation.navigate("Edit Inventory", {
            name: "Bags of senegal beans",
            price: 60000,
            description: "Bags of senegal beans due for delivery and sale",
            stock: 3,
          })
        }
      />
    </Screen>
  );
};

export default InventoryListingScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#F1F3F4",
    padding: 20,
  },
});
