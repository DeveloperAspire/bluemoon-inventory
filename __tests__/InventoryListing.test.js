import React from "react";
import renderer from "react-test-renderer";
import InventoryListingScreen from "../app/screens/InventoryListingScreen";

test("renders correctly", () => {
  const tree = renderer.create(<InventoryListingScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
