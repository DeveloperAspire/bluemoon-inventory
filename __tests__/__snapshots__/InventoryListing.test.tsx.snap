// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flex": 1,
        "paddingTop": 54,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        {
          "backgroundColor": "#F1F3F4",
          "flex": 1,
          "padding": 20,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
            "fontFamily": "Mulish_400Regular",
            "fontSize": 18,
            "paddingHorizontal": 5,
            "paddingVertical": 2,
          },
          undefined,
          {
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      You don't have a saved inventory yet! ðŸ˜ƒ
    </Text>
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
